Sources/GRPC/GRPCServerRequestRoutingHandler.swift

```diff
@@ -216,25 +216,59 @@ extension GRPCServerRequestRoutingHandler: ChannelInboundHandler, RemovableChann
     }
   }
 
+    /// A call URI split into components.
+    struct CallPath {
+        /// The name of the service to call.
+        var service: String.UTF8View.SubSequence
+        /// The name of the method to call.
+        var method: String.UTF8View.SubSequence
+
+        /// Charater used to split the path into components.
+        private let pathSplitDelimiter = "/".utf8.first!
+
+        /// Split a path into service and method.
+        /// Split is done in UTF8 as this turns out to be approximately 10x faster than a simple split.
+        /// URI format: "/package.Servicename/MethodName"
+        init?(requestUri: String) {
+            let requestUriUTF8 = requestUri.utf8
+            let maybeFirstIndex = requestUriUTF8.firstIndex(of: pathSplitDelimiter)
+            guard let firstIndex = maybeFirstIndex else {
+                return nil
+            }
+            let afterFirstDelimiter = requestUriUTF8[requestUriUTF8.index(after:firstIndex)...]
+            if let secondIndex = afterFirstDelimiter.firstIndex(of: pathSplitDelimiter) {
+                self.service = afterFirstDelimiter[..<secondIndex]
+                let afterSecondDelimiter = afterFirstDelimiter[afterFirstDelimiter.index(after: secondIndex)...]
+                if let thirdIndex = afterSecondDelimiter.firstIndex(of: pathSplitDelimiter) {
```

Comment:
All of this seems like it might be nicer wrapped up in a helpful extension. Perhaps:

```swift
extension Collection where SubSequence == Self {
    /// Trims out the prefix up to separator, and returns it. Sets self to the subsequence
    /// after the separator, and returns the subsequence before the separator. Returns nil
    /// and does not mutate self if `separator` is not present.
    mutating func trim(through separator: Element) -> SubSequence? {
        if let separatorIndex = self.firstIndex(of: separator) {
            let indexAfterSeparator = self.index(after: separatorIndex)
            defer { self = self[indexAfterSeparator...] }
            return self[..<separatorIndex]
        } else {
            return nil
        }
    }
}
```

This will compose pretty nicely, I think, and greatly clean up this call site.